ee/api/license.py

```diff
@@ -41,3 +45,22 @@ def get_queryset(self) -> QuerySet:
             return License.objects.none()
 
         return super().get_queryset()
+
+    def destroy(self, request: request.Request, pk=None, **kwargs) -> Response:
+        license = get_object_or_404(License, pk=pk)
+        validation = requests.post("https://license.posthog.com/licenses/deactivate", data={"key": license.key})
+        validation.raise_for_status()
+
+        teams = Team.objects.exclude(is_demo=True).order_by("pk")[1:]
+        for team in teams:
+            team.delete()
+
+        #  delete any organization where we've deleted all teams
+        # there is no way in the interface to create multiple organizations so we won't bother informing people that this is happening
+        for organization in Organization.objects.all():
+            if organization.teams.count() == 0:
+                organization.delete()
```

Comment:
Agree on locking out rather than deleting projects.  That way if they come back they haven't lost anything.
We may also have the (very occasional) case where someone goes from Enterprise -> Scale so we need to remove the enterprise key (and project permissions) but they still keep multi projects and premium features.

Reply:
> Agree on locking out rather than deleting projects. That way if they come back they haven't lost anything.

I doubt many (high value) customers will cancel and then come back. I think the risk of them getting around the lock are too big + it adds more unnecessary complexity. I think cancelling your license should be a bit painful.

>We may also have the (very occasional) case where someone goes from Enterprise -> Scale so we need to remove the enterprise key (and project permissions) but they still keep multi projects and premium features.

We can solve for this case while still deleting project if people don't have a valid license